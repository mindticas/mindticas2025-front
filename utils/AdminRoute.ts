import Layout from '@/components/Layout';
import { Box } from '@chakra-ui/react';

export class AdminRoute {
    // Fixed route for admin
    private static route: string = '/admin';

    // Returns the Layout component if the pathname contains '/admin', otherwise returns box
    static getLayout(pathname: string) {
        return pathname.includes(this.route) ? Layout : Box;
    }
}
