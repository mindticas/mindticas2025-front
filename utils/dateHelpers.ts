// Gets the name of the day according to the full calendar format
export const getDayName = (dateStr: string): string => {
    const [year, month, day] = dateStr.split('-').map(Number)
    const date = new Date(year, month -1, day)
    return date.toLocaleDateString('es-ES', { weekday: 'long'})
}
// Get the milliseconds in a day
export function addDays(date: Date, days: number): Date {
    const milisecondsPerDay = 24 * 60 * 60 * 1000; 
    return new Date(date.getTime() + days * milisecondsPerDay)
}