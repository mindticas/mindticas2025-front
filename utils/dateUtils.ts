import { DateTime } from "luxon";

interface DateTimeParts {
    date: string;
    time: string;
}

// * @returns Object with (YYYY-MM-DD) y time (HH:MM)

export const splitDateTimeFromISO = (
    isoString: string | null | undefined,
): DateTimeParts => {
    if (!isoString) return { date: '', time: '' };

    const dateObj = new Date(isoString);

    if (isNaN(dateObj.getTime())) {
        return { date: '', time: '' };
    }

    // format date (YYYY-MM-DD)
    const date = DateTime.fromJSDate(dateObj)
       .setZone('local')
        .toFormat('yyyy-MM-dd');

    // format time (HH:MM)
    const time = dateObj
        .toLocaleTimeString('es-ES', {
            hour: '2-digit',
            minute: '2-digit',
            hour12: false,
        })
        .replace(/\./g, '');

    return { date, time };
};
