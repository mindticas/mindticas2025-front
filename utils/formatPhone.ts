export default function formatPhone(phone: string): string {
    // Remove everything that is not digits
    const digits = phone.replace(/\D/g, '');
    // Applies format only if it has 10 digits
    if (digits.length === 10) {
        return `${digits.slice(0, 3)}-${digits.slice(3, 6)}-${digits.slice(6)}`;
    }
    return phone; // Returns the original if it doesn't match
}
