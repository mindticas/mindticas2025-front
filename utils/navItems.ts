import { BarChart, Calendar, Scissors, UserCog, Users } from 'lucide-react';

export const navItems = [
    { name: 'Citas', href: '/admin/', icon: Calendar },
    { name: 'Clientes', href: '/admin/clients', icon: Users },
    { name: 'Empleados', href: '/admin/employees', icon: UserCog },
    { name: 'Tratamientos', href: '/admin/treatments', icon: Scissors },
    { name: 'Reportes', href: '/admin/reports', icon: BarChart },
];
