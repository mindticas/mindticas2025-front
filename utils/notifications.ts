import { toaster } from '@/components/ui/toaster';

type ToastType = 'success' | 'error' | 'info' | 'warning';

interface ToastOption {
    duration?: number;
    title: string;
    description?: string;
}

const showToast = (type: ToastType, options: ToastOption) => {
    toaster.create({
        type,
        duration: options.duration || 5000,
        title: options.title,
        description: options.description,
    });
};

const TreatmentNotification = {
    createSuccess: () =>
        showToast('success', { title: 'Tratamiento creado con éxito' }),
    updateSuccess: () =>
        showToast('success', { title: 'Tratamiento actualizado con éxito' }),
    deleteSuccess: () =>
        showToast('success', { title: 'Tratamiento eliminado con éxito' }),
    createError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al crear el tratamiento',
        }),
    updateError: (message?: string) =>
        showToast('error', {
            title: message || 'Error al actualizar el tratamiento0',
        }),
    deleteError: () =>
        showToast('error', { title: 'Error al eliminar el tratamiento' }),
    noChanges: () =>
        showToast('info', {
            title: 'No se detectaron cambios para actualizar',
        }),
};

export default TreatmentNotification;
