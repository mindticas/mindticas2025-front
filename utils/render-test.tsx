import { Provider } from '@/components/ui/provider';
import { BookingProvider } from '@/context/BookingContext';
import { render as rtlRender } from '@testing-library/react';

export function render(ui: React.ReactNode) {
    return rtlRender(<>{ui}</>, {
        wrapper: (props: React.PropsWithChildren) => (
            <Provider>
                <BookingProvider>{props.children}</BookingProvider>
            </Provider>
        ),
    });
}
