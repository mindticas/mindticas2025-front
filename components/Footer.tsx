import React from 'react';
import { Box, Flex, Text, Container, Icon, Spinner } from '@chakra-ui/react';
import { MapPin, Phone } from 'lucide-react';
import { useBusiness } from '@/context/BusinessContext';
import formatPhone from '@/utils/formatPhone';
import Link from 'next/link';

export default function Footer() {
    const { businessInfo, isLoading } = useBusiness();
    const getDate = new Date();
    const currentYear = getDate.getFullYear();
    return (
        <footer>
            <Box as='footer' bg='black' color='white' py={8} mt={6}>
                <Container maxW='container.2xl' px={4}>
                    <Flex
                        direction={{ base: 'column', md: 'row' }}
                        justify='space-between'
                        align='center'
                        gap={5}
                        px={4}
                    >
                        <Flex
                            direction='column'
                            align={{ base: 'center', md: 'flex-start' }}
                            mb={{ base: 4, md: 0 }}
                        >
                            <Flex
                                align='center'
                                mb={2}
                                w='100%'
                                justifyContent='space-between'
                            >
                                <Flex>
                                    <img
                                        src='/logo.png'
                                        width={30}
                                        height={90}
                                        alt='Dynamic Image'
                                    />
                                    <Text fontSize='xl' fontWeight='bold'>
                                        Elegangster
                                    </Text>
                                </Flex>
                            </Flex>

                            <Flex align='center'>
                                <Icon as={MapPin} h={4} w={4} />
                                {isLoading ? (
                                    <Spinner
                                        size='xs'
                                        color='gray.300'
                                        ml={2}
                                    />
                                ) : (
                                    <Text fontSize='sm' ml={2}>
                                        {businessInfo.address}
                                    </Text>
                                )}
                            </Flex>

                            <Flex align='center' mt={1}>
                                <Icon as={Phone} h={4} w={4} />
                                {isLoading ? (
                                    <Spinner
                                        size='xs'
                                        color='gray.300'
                                        ml={2}
                                    />
                                ) : (
                                    <Text fontSize='sm' ml={2}>
                                        {formatPhone(businessInfo.phone)}
                                    </Text>
                                )}
                            </Flex>
                        </Flex>
                        <Flex direction='column' gap={4}>
                            {!isLoading && businessInfo.instagram && (
                                <Flex align='center' gap={4}>
                                    <Text>Redes sociales: </Text>
                                    <Link
                                        target='_blank'
                                        href={businessInfo.instagram}
                                        passHref
                                    >
                                        <img
                                            src='/instagram-w.png'
                                            width={25}
                                            height={85}
                                            alt='Instagram'
                                        />
                                    </Link>
                                </Flex>
                            )}
                            <Box textAlign={{ base: 'center', md: 'right' }}>
                                <Text>
                                    &copy; {currentYear}{' '}
                                    {isLoading ? (
                                        <Spinner
                                            size='xs'
                                            color='gray.300'
                                            ml={2}
                                        />
                                    ) : (
                                        businessInfo.name
                                    )}
                                    . Todos los derechos reservados.
                                </Text>
                            </Box>
                        </Flex>
                    </Flex>
                </Container>
            </Box>
        </footer>
    );
}
