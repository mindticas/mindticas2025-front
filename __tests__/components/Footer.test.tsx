import Footer from '@/components/Footer';
import { render } from '@/utils/render-test';

describe('Calendar Snapshot', () => {
    it('should match the snapshot', () => {
        const { asFragment } = render(<Footer />);
        expect(asFragment()).toMatchSnapshot();
    });
});
