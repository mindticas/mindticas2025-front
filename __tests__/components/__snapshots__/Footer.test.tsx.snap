// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calendar Snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <script
    nonce=""
  >
    ((e,s,n,m,o,l,u,h)=&gt;{let a=document.documentElement,v=["light","dark"];function p(c){(Array.isArray(e)?e:[e]).forEach(y=&gt;{let k=y==="class",S=k&&l?o.map(f=&gt;l[f]||f):o;k?(a.classList.remove(...S),a.classList.add(c)):a.setAttribute(y,c)}),C(c)}function C(c){h&&v.includes(c)&&(a.style.colorScheme=c)}function d(){return window.matchMedia("(prefers-color-scheme: dark)").matches?"dark":"light"}if(m)p(m);else try{let c=localStorage.getItem(s)||n,y=u&&c==="system"?d():c;p(y)}catch(c){}})("class","theme","system",null,["light","dark"],null,true,true)
  </script>
  <footer>
    <footer
      class="css-111rync"
    >
      <div
        class="chakra-container css-t10ba0"
      >
        <div
          class="css-a6cv6e"
        >
          <div
            class="css-11lq829"
          >
            <div
              class="css-1cn684i"
            >
              <img
                alt="Dynamic Image"
                height="90"
                src="/logo.png"
                width="30"
              />
              <p
                class="css-pntfz2"
              >
                Elegangster
              </p>
            </div>
            <div
              class="css-70qvj9"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-map-pin chakra-icon css-1q0syye"
                fill="none"
                focusable="false"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 10c0 4.993-5.539 10.193-7.399 11.799a1 1 0 0 1-1.202 0C9.539 20.193 4 14.993 4 10a8 8 0 0 1 16 0"
                />
                <circle
                  cx="12"
                  cy="10"
                  r="3"
                />
              </svg>
              <p
                class="css-c96f87"
              >
                Av. Real Bugambilias 286, Lomas de la Higuera
              </p>
            </div>
            <div
              class="css-8kg6q2"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-phone chakra-icon css-1q0syye"
                fill="none"
                focusable="false"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 16.92v3a2 2 0 0 1-2.18 2 19.79 19.79 0 0 1-8.63-3.07 19.5 19.5 0 0 1-6-6 19.79 19.79 0 0 1-3.07-8.67A2 2 0 0 1 4.11 2h3a2 2 0 0 1 2 1.72 12.84 12.84 0 0 0 .7 2.81 2 2 0 0 1-.45 2.11L8.09 9.91a16 16 0 0 0 6 6l1.27-1.27a2 2 0 0 1 2.11-.45 12.84 12.84 0 0 0 2.81.7A2 2 0 0 1 22 16.92z"
                />
              </svg>
              <p
                class="css-c96f87"
              >
                (312) 291-3365
              </p>
            </div>
          </div>
          <div
            class="css-17btmjg"
          >
            <p
              class="css-0"
            >
              Â© 2025 Elegangster Barber Studio. Todos los derechos reservados.
            </p>
          </div>
        </div>
      </div>
    </footer>
  </footer>
</DocumentFragment>
`;
